internal/provider/resource_group.go

```diff
@@ -76,6 +76,19 @@ func setFromGroupResponseMap(d *schema.ResourceData, raw map[string]interface{})
 }
 
 func resourceGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		apiResponse map[string]interface{}
+		errs        diag.Diagnostics
+	)
+	defer func() {
+		if apiResponse == nil {
+			return
+		}
+		if err := setFromGroupResponseMap(d, apiResponse); err != nil {
+			errs = append(errs, diag.FromErr(err)...)
+		}
+	}()
```

Comment:
Putting this defer at the top is very confusing to me. Could we put it on line 124 and remove the conditional check on a capture variable? You're also going to want to add a named variable if you want to mutate the response:

```go
func resourceGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) (errs diag.Diagnostics) {
...
if gcr == nil {
	return diag.Errorf("group nil after create")
}
defer func() {
	if err := setFromGroupResponseMap(d, apiResponse); err != nil {
		errs = append(errs, diag.FromErr(err)...)
	}
}()
```